pkg/resources/virtual/common/common.go

```diff
@@ -25,25 +24,29 @@ type DefaultFields struct {
 }
 
 // GetTransform produces the default transformation func
-func (d *DefaultFields) GetTransform() cache.TransformFunc {
-	return d.transform
+func (d *DefaultFields) GetTransform() func(obj *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+	return d.TransformCommon
 }
 
-// transform implements virtual.VirtualTransformFunc, and adds reserved fields/summary
-func (d *DefaultFields) transform(obj any) (any, error) {
-	raw, isSignal, err := getUnstructured(obj)
+func (d *DefaultFields) GetUnstructuredObjectWrapper(obj interface{}) (interface{}, bool, error) {
```

Comment:
Nitpick: this is only ever used inside of GetTransformFunc. I suggest inlining (using `GetUnstructured` directly) there.

Reply:
Yup, true